packages/data/README.md

```diff
@@ -251,6 +251,45 @@ Specific implementation differences from Redux and React Redux:
 
 <!-- START TOKEN(Autogenerated API docs) -->
 
+<a name="AsyncModeProvider" href="#AsyncModeProvider">#</a> **AsyncModeProvider**
+
+Context Provider Component used to switch the data module component rerendering
+between Sync and Async modes.
+
+_Usage_
+
+```js
+import { useSelect, AsyncModeProvider } from '@wordpress/data';
+
+function BlockCount() {
+  const count = useSelect( ( select ) => {
+    return select( 'core/block-editor' ).getBlockCount()
+  } );
+
+  return count;
+}
+
+function App() {
+  return (
+    <AsyncModeProvider value={ true }>
+      <BlockCount />
+    </AsyncModeProvider>
+  );
+}
+```
+
+In this example, the BlokcCount component is rerendered asynchronously.
+It means if a more critical task is being performed (like typing in an input),
+the rerendering is delayed until the browser becomes IDLE.
+
```

Comment:
Can we add a note that you can opt-out from the async mode at any level of nesting by overriding the value with another component used? After writing it, I realized It might be a challenge to explain 😄 

Unless you can't opt-out once you declared the sub-tree renders async?